### PR TITLE
Add Tests

### PR DESCRIPTION
**Pull request by brianwchou**
### Description
Create tests for repo
- [x] advanceTime
- [x] advanceBlock
- [x] advanceTimeAndBlock
- [x] takeSnapShot (takeSnapShot is currently spelled *takeSnapshot*
- [x] revertToSnapShot

### CODE REVIEW

File: truffle-config.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    // You should run a client (like ganache-cli, geth or parity) in a separate terminal
+    // tab if you use this network and you must also set the `host`, `port` and `network_id`
+    // options below to some value.
+    //
+     development: {
+      host: "127.0.0.1",     // Localhost (default: none)
+      port: 8545,            // Standard Ethereum port (default: none)
+      network_id: "*",       // Any network (default: none)
+     },
+
```
Comment (brianwchou) [inline truffle-config.js:50]: Since this testing module only works with ganache shouldn't we just use what is given by default instead of leaving the port open? I'm going to set truffle as a development dependency but I'm not sure if its necessary to add ganache-cli as well.

Comment (ejwessel) [reply truffle-config.js:50]: Sure, but that's unrelated to this PR. Open a new issue for it.

**Comment by brianwchou** [comment]
Related to advancingTime function:
https://ethereum.stackexchange.com/questions/27447/evm-increasetime-does-not-increase-timestamp

File: test/utils.test.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        const advanceBlockNum = advancedBlock.number
+        
+        assert.equal(currentBlockNum + 1, advanceBlockNum, "New block was not mined")
+    })
+
+    it("Test advanceTime", async() => {
+        const currentBlock = await web3.eth.getBlock('latest')
+
+        await helper.advanceTime(SECONDS_IN_DAY);
+        await helper.advanceBlock()
```
Comment (ejwessel) [inline test/utils.test.js:25]: I think you should explain why `advanceBlock()` is needed with a comment

File: test/utils.test.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        await helper.advanceTimeAndBlock(SECONDS_IN_DAY)
+
+        const advancedBlock = await web3.eth.getBlock('latest')
+        assert.isBelow(currentBlock.timestamp, advancedBlock.timestamp, "Time and Block was not advanced")
+    })
+
+    it("Test takeSnapShot", async() => {
+        const snapShot = await helper.takeSnapshot()
+        const snapShotId = snapShot.result
+
```
Comment (ejwessel) [inline test/utils.test.js]: remove new line

**Review by ejwessel — CHANGES_REQUESTED** [review]
I like that you've separated the logic with new lines, but can we add some comments that describe those logic blocks?

**Comment by ejwessel** [comment]
I messed up and wasn't paying attention to my PRs and merged package into yours. Apologies.

File: test/utils.test.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        const advanceBlockNum = advancedBlock.number
+        
+        assert.equal(currentBlockNum + 1, advanceBlockNum, "New block was not mined")
+    })
+
+    it("Test advanceTime", async() => {
+        const currentBlock = await web3.eth.getBlock('latest')
+
+        await helper.advanceTime(SECONDS_IN_DAY);
+        await helper.advanceBlock() // advanced time does not show unless block has been mined and shown in new block
```
Comment (ejwessel) [inline test/utils.test.js]: `advanced time does not show unless block has been mined and shown in new block`
Change to
`time doesn't update unless a block is mined`
Place the comment above this line.

File: test/utils.test.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    })
+
+    it("Test takeSnapshot", async() => {
+        const snapshot = await helper.takeSnapshot()
+        const snapshotId = snapshot.result
+        assert.exists(snapshotId, "Unable to produce snapshot")
+    })
+
+    it("Test revertToSnapShot", async() => {
+        // initial data captured
```
Comment (ejwessel) [inline test/utils.test.js]: Not specific enough

File: test/utils.test.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        const snapshot = await helper.takeSnapshot()
+        const snapshotId = snapshot.result
+        const currentBlock = await web3.eth.getBlock('latest')
+
+        // time advanced to simulated that other transactions occured over a day
```
Comment (ejwessel) [inline test/utils.test.js]: Too wordy. You're moving time forward.

**Comment by ejwessel** [comment]
Comments are good but they should be clearer and more concise. You should be able to look over the comments and know what the blocks do.

File: test/utils.test.js (hunk 6 above)
Comment (brianwchou) [reply test/utils.test.js]: It's hard to tell when to comment. If i say something simple that the code already describes, it feels redundant. Like here in lines 50 - 52 don't really need me to say that time is moved forward.

File: test/utils.test.js (hunk 5 above)
Comment (brianwchou) [reply test/utils.test.js]: Here also maybe we don't need to comment? The test already describes itself

File: test/utils.test.js (hunk 4 above)
Comment (brianwchou) [reply test/utils.test.js]: I agree with this

File: test/utils.test.js (hunk 6 above)
Comment (ejwessel) [reply test/utils.test.js]: > It's hard to tell when to comment. If i say something simple that the code already describes, it feels redundant. Like here in lines 50 - 52 don't really need me to say that time is moved forward.

It's true. It just takes some practice. 
It's a combination of using comments, breaking up the code into logical sections, and naming.

**Review by ejwessel — APPROVED** [review]
@brianwchou  Take a look at what I've done in the last commit. Take a look at the visual grouping, naming, and where I've placed comments. 

I will admit, even though I feel comfortable with it. I'm still not 100%. But this suffices.
